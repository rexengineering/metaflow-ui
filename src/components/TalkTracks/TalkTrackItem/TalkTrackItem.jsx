import React from "react";
import { Button, Chip, makeStyles, Paper, Typography } from "@material-ui/core";
import { talkTrackItemShape } from "../../../utils/shapes";
import clsx from "clsx";

const useStyles = makeStyles((theme) => ({
  paper: {
    background: theme.palette.background.default,
    padding: theme.spacing(2),
  },
  title: {
    marginBottom: theme.spacing(1),
  },
  speech: {
    marginBottom: theme.spacing(2),
  },
  actions: {
    display: "flex",
    justifyContent: "left",
  },
  action: {
    margin: theme.spacing(0, 1, 2, 0),
    textTransform: "uppercase",
    lineHeight: 1,
  },
  buttons: {
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
    marginTop: theme.spacing(2),
  },
}));

function TalkTrackItem({ identifier, title, speech, actions, onActionSelected, onSkip, onContinue, className }) {
  const classes = useStyles();
  return (
    <Paper elevation={0} className={clsx(classes.paper, className)}>
      <Typography color="primary" className={classes.title} variant="body1">
        {title}
      </Typography>
      <Typography className={classes.speech} variant="body2">
        {speech}
      </Typography>

      {actions.length ? (
        <section>
          <Typography color="primary" className={classes.title} variant="body1">
            Select button based on customer inquiry
          </Typography>
          <section className={classes.actions}>
            {actions.map(({label, talktrackId}) => (
              <Chip
                key={`${label}-${talktrackId}`}
                className={classes.action}
                variant="outlined"
                onClick={() => onActionSelected(talktrackId)}
                size="small"
                label={label}
              />
            ))}
          </section>
        </section>
      ) : null}
      <section className={classes.buttons}>
        <Button type="button" onClick={() => onSkip(identifier)}>
          Skip
        </Button>
        <Button type="button" onClick={() => onContinue(identifier)}>
          Continue
        </Button>
      </section>
    </Paper>
  );
}

TalkTrackItem.defaultProps = {
  actions: [],
  onActionSelected: () => {},
  onContinue: () => {},
  active: false,
  className: "",
};

TalkTrackItem.propTypes = talkTrackItemShape;

export default TalkTrackItem;
