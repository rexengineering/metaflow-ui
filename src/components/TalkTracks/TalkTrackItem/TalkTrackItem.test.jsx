import React from "react";
import { render, screen } from "@testing-library/react";
import TalkTrackItem from "./TalkTrackItem";

describe("<TalkTrackItem />", () => {
  const defaultProps = {
    identifier: "16515",
    title: "Intro",
    speech: "Some speech",
    actions: [],
    onSkip: () => {},
    onActionSelected: () => {},
    onContinue: () => {},
  };

  const renderTalkTrack = ({
    title,
    speech,
    actions,
    onSkip,
    onActionSelected,
    onContinue,
    identifier
  } = defaultProps) => (
    <TalkTrackItem
        identifier={identifier}
        speech={speech}
        title={title}
        onContinue={onContinue} onActionSelected={onActionSelected}
        onSkip={onSkip}
        actions={actions}
    />
  );

  it("should render", () => {
    const { asFragment } = render(renderTalkTrack());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderTalkTrack());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should render all passed actions", () => {
    const actions = [ {label: "Action1", talktrack_id: "action1"},  {label: "action2", talktrack_id: "action2"},  {label: "Action3", talktrack_id: "action3"}];
    render(renderTalkTrack({ ...defaultProps, actions }));
    actions.forEach(({label}) =>
      expect(screen.getByText(label)).toBeInTheDocument()
    );
  });
});
