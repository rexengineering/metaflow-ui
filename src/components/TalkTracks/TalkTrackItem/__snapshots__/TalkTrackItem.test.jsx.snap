// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TalkTrackItem /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root makeStyles-paper-7 MuiPaper-elevation0 MuiPaper-rounded"
  >
    <p
      class="MuiTypography-root makeStyles-title-8 MuiTypography-body1 MuiTypography-colorPrimary"
    >
      Intro
    </p>
    <p
      class="MuiTypography-root makeStyles-speech-9 MuiTypography-body2"
    >
      Some speech
    </p>
    <section
      class="makeStyles-buttons-12"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Skip
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Continue
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </section>
  </div>
</DocumentFragment>
`;
